synthesizer/process/src/cost.rs

```diff
@@ -19,9 +19,23 @@ use console::{
     prelude::*,
     program::{FinalizeType, Identifier, LiteralType, PlaintextType},
 };
-use ledger_block::{Deployment, Execution};
+use ledger_block::{Deployment, Execution, Transaction};
 use synthesizer_program::{CastType, Command, Finalize, Instruction, Operand, StackProgram};
 
+/// Returns the compute cost for a transaction in microcredits.
+/// This is used to limit the amount of compute in the block generation hot
+/// path. This does NOT represent the full costs which a user has to pay.
+pub fn compute_cost<N: Network>(process: &Process<N>, transaction: &Transaction<N>) -> Result<u64> {
+    match transaction {
+        // Synthesis cost accounts for the majority of deployment transaction compute.
+        Transaction::Deploy(_, _, _, deployment, _) => deployment_synthesis_cost(deployment),
```

Comment:
I realized now that `TRANSACTION_SPEND_LIMIT` was based on a maximum spent on finalize, but now we are also considering for `deployment_synthesis_cost`.

Can you estimate how this affects the bounds? How much would a maximum contraint size deployment fill up? How many deployments can we expect in a proposal, etc.

Reply:
In that vein, it would be good to add metrics like how many "transfer_publics" could fill a proposal. (Obviously, this is prior to fee reductions).

Reply:
I've added a test case, hopefully this is what you had in mind 🤞

Reply:
Curious how much of the proposal spend limit does a max size deployment take up?

Reply:
`BATCH_SPEND_LIMIT / max_synthesis_cost` in the test I added is around 1.9, this is without the storage cost so it currently would take up most of the proposal (as we don't pack the remaining batch/block space.